quaterion/train/xbm/xbm_config.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+
+class XbmDevice(str, Enum):
+    """Device selection for placement of the buffer"""
+
+    CPU = "cpu"
+    """Buffer created in CPU"""
+
+    CUDA = "cuda"
+    """Buffer created in GPU"""
+
+    AUTO = "auto"
+    """Buffer created in GPU if available. In CPU, otherwise."""
+
+
+@dataclass
+class XbmConfig:
+    """Determine XBM settings.
+
+    This class should be returned from
+    :meth:`~quaterion.train.trainable_model.TrainableModel.configure_xbm`
+    """
+
+    embedding_size: int
+    """Dimensionality of embeddings outputted by the encoder"""
+
+    weight: Optional[float] = 1.0
```

Comment:
It's the default value in their implementation. Most often, this will be combined with a scheduler to decay LR, but I chose to simplify the example with a smaller weight value.